pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +116,189 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Read the status conditions and assign some variables
+	provisioning := condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupProvisioning)
+	completed := condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
+
+	// If the status is not yet provisioning or completed, check for automatic or manual approval
+	if !provisioning && !completed {
+		userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+		if err != nil {
+			err := r.setStatusFailedToReadUserApprovalPolicy(instance, "Failed to read ConfigMap value " + config.ToolchainConfigMapUserApprovalPolicy)
+			if err != nil {
+				reqLogger.Error(err, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+				return reconcile.Result{}, err
+			}
+
+			reqLogger.Error(err, "Error reading user approval policy")
+			return reconcile.Result{}, err
+		}
+
+		// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+		// then proceed with the signup
+		if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
```

Comment:
Could you please break down this whole method (`Reconcile`) to smaller methods/functions? It is really long and hard to orientate in

Reply:
I've moved the code which creates the MUR into a separate function.